lib/core/src/Cardano/Wallet/Primitive/CoinSelection/Random.hs

```diff
@@ -130,6 +131,33 @@ random opt outs utxo = do
   where
     guard = except . left ErrInvalidSelection . validate opt
 
+randomWithoutTxOut
+    :: forall m e. MonadRandom m
+    => CoinSelectionOptions e
+    -> UTxO
+    -> ExceptT (ErrCoinSelection e) m (CoinSelection, UTxO)
+randomWithoutTxOut opt utxo = do
+    let nUtxo = L.length $ (Map.toList . getUTxO) utxo
+    let maxN = fromIntegral $ maximumNumberOfInputs opt 0
+
+    when (nUtxo == 0) $ throwE ErrInputsDepleted
+
+    randomMaybe <- lift $ runMaybeT $
+        pickRandomT utxo >>= \(io, utxo') -> do
+            if (L.length io > maxN) then
+                MaybeT $ return Nothing
+            else do
+                let (_, TxOut _ c) = io
+                MaybeT $ return $ Just (CoinSelection [io] [] [c], utxo')
+
+    case randomMaybe of
+        Just (sel, remUtxo) -> do
+            guard sel $> (sel, remUtxo)
+        Nothing ->
+            throwE $ ErrMaximumInputsReached (fromIntegral maxN)
+  where
+    guard = except . left ErrInvalidSelection . validate opt
```

Comment:
I might be missing something but, why is it needed to define a new function `randomWithoutTxOut` instead of just adjusting the existing `random` to take a list of outputs (instead of a `NonEmpty` list)? 

Also, it seems that this version does not fallback anymore to largest first? It that on purpose?

Reply:
Frankly speaking I tried to accommodate `[TxOut]` in random, but `makeSelection` and `improve` make it difficult for me to do it. mainly because of heavy use `TargetRange` inside them :thinking: